design/codebases.go

```diff
@@ -63,15 +63,29 @@ var workspace = a.Type("Workspace", func() {
 
 var workspaceAttributes = a.Type("WorkspaceAttributes", func() {
 	a.Description(`JSONAPI store for all the "attributes" of a workspace. +See also see http://jsonapi.org/format/#document-resource-object-attributes`)
+	a.Attribute("id", d.String, "The workspace id", func() {
+		a.Example("workspacephly8l91tqg0u08x")
+	})
 	a.Attribute("name", d.String, "The workspace name", func() {
 		a.Example("test")
 	})
-	a.Attribute("description", d.String, "The URL of the codebase ", func() {
+	a.Attribute("description", d.String, "The workspace description", func() {
 		a.Example("")
 	})
 	a.Attribute("status", d.String, "The workspace status", func() {
 		a.Example("STOPPED");
 	})
+	a.Attribute("links", links, "The workspace links")
```

Comment:
if the `id`, `name` and `status` attributes are mandatory, you could use the `a.Required(...)` to mark them as such.